### DISCUSSION TITLE
Proper rotating and trimming of source pages

### DISCUSSION THREAD

**Discussion by lsces** [pdfarranger/pdfarranger#1140] [General]
This was started as a different discussion, but as various thoughts were progressed I realised that I was looking at the problem in the wrong way. While what we are seeing on the screen is just what we want to see, the raw pages are not being processed to produce new pages, but rather simply clipped and rotated as required? With the wrong tools such as an ebook reader, the cleaver formatting added by PDF Arranger is lost at least partially. Having created a correct view of the set of images, is there any way to then produce a clean set of images trimmed and rotated properly? I am thinking that these can then be processed to do a full OCR translation to remove the need for the images at all, but that again is probably not necessary. Magazines I have been processing in the past via Finereader result in a substantial amount of pure formatted text interleaved with smaller picture areas as required and the final pdf is a lot smaller than a simple scanned document with a searchable text overlay ... with white background rather than the faded paper view ... and images that are individually selectable.

**Comment by kbengs** [comment]
> but rather simply clipped and rotated as required? 

Yes page is not rotated but only rotation information is set and page is cropped by setting the mediabox. There is also cropbox and trimbox which pdfarranger deletes. In all PDF viewers I have tried the resulted PDF looks right.

> an ebook reader, the cleaver formatting added by PDF Arranger is lost at least partially.

Do you mean that you see pages unrotated and uncropped?

> is there any way to then produce a clean set of images trimmed and rotated properly?

No PDF Arranger currently can not partly delete a page.

**Reply by lsces** [reply]
That is where I think I had got to ... 
YES the ebook reader displays an A4 page, but it's image is the whole A3 portrait scan. It is not what I would normally use to view pdf documents, but I am using calibre to catalogue the library and that switches ebook reader in by default.
Just to complete my knowledge ... a scan will be deleted if it is no longer used? How about ordering of additional scans when they are added. I am often reviewing the source material and if an A3 spread is skewed too much or has messed up I will rescan the problem pages and add them back in so that my raw A3 view has everything in the right order ... and I save that file from PDF Arranger. 
I WILL spend more time looking at the detail, but currently I need to concentrate on scanning the source material.

**Reply by kbengs** [reply]
Yes that's right, if a page is deleted it will not be in the saved PDF.

**Reply by lsces** [reply]
I should probably have been taking a little more care on how the raw images are being scanned and going forward I do need to extract pictures that are the right orientation, but that PDF Arrange is NOT actually doing anything to the original images is the useful thing here.  All I perhaps need to be aware of is that a dump of the raw images may ignore any arrangement that is contained IN the PDF Arrange view of those images? Hence the thought about 'printing' a set of processed A4 pages from the stored A3 images ... and that rotating a miss-scanned page does nothing to the image :)

**Reply by kbengs** [reply]
> a dump of the raw images may ignore any arrangement that is contained IN the PDF Arrange view of those images?

Yes. If you try to extract the image with `Extract - Copy Image`, then paste to image editing software, it will be a unrotated full sized image.

**Comment by dreua** [comment]
I think what you want to do is beyond the scope of PDF Arranger, sorry.

**Reply by lsces** [reply]
Yes ... and No
I am not expecting PDF Arranger to do anything beyond 'arranging the pages' ... OCR is a separate step, and it may be that what I am looking for is another script which takes the PDF Arranger output and 'prints' it as a set of individual A4 pages rotated to the right orientation? Is that a step that PDF Arranger COULD provide? I think so and it is something I will look at myself in the future as it will tidy up the thousands of 'A4' files I have saved using PDF Arranger. 
I do need to dig into ocrmypdf as well to see what it's outputting, but the ebook reader is looking at the OCRed version of the file.
